### ISSUE TITLE
Target the 16 years old people

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1111]
Note: no one of these checkboxes are specific to the 16yo people, they are meant for everyone, and 16yo people is part of every people. We must get those checkboxes fixed for everyone so we can target 16yo people which _must be the target audience_.

- [x] ask for nickname on first startup \
  https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/pull/14
- [x] open server list at startup
- [ ] list lan servers above net servers in a single list \
  https://github.com/Unvanquished/Unvanquished/issues/1112
- [x] usable default key bindings \
  https://github.com/Unvanquished/Unvanquished/issues/544
- [x] mouse wheel immediate weapon cycling, including grenade (launch with fire) and medkit (activate with fire), grenade and medkit removable from weapon cycle for power users \
  https://github.com/Unvanquished/Unvanquished/issues/544
- [x] explicit bind entry in configuration window to switch primary weapon, secondary weapon, grenade \
  grenade is not included, but we now have mouse roll for switching weapons (which is told by tutorial): \
  https://github.com/Unvanquished/Unvanquished/pull/1182
  grenade has its own bind and is told by tutorial.
- [x] find a better buildable removal mechanism not involving two key binds
- [ ] wallwalk interpolation: \
  https://github.com/Unvanquished/Unvanquished/issues/609
- [x] tell people the game is a strategy game in first person view, player don't win by killing most enemies or carrying a flag to base but by denying the other the right to respawn, checkmate!
- [x] reference tremulous in obscure places for those dinosaurs having lived that age, no more \
 https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/pull/36 \
 https://github.com/Unvanquished/Unvanquished/pull/1879

We must target 16 years old people. Let's imagine the game is neither alpha neither beta in two years, it means the people being 16 years old at this time are 14 years old today.

Our current alphas today must target 16 years old people today. People having 16 years old today were 3 years old when Tremulous 1.1 came out, they were learning how to talk and walk.

And for the years coming, we must remind we target the people that are currently learning how to talk and walk so they can have a game when they reach 16.

**Comment by kentone** [comment]
As for the build system, you could use left click to build, right click to unbuild, on Squad is like that by example, you could set that functions to the build gun

**Comment by kentone** [comment]
Also a problem I see with the game is the population, if you enter and you see 0 players online, there's no motivation to play, maybe some propaganda could be done to the game, maybe place it on GOG or itch.io (Steam would be optimal speaking about users, but I don't think the terms of use would match this game :( )

**Comment by illwieckz** [comment]
> As for the build system, you could use left click to build, right click to unbuild, on Squad is like that by example, you could set that functions to the build gun

Not a bad idea.

Btw we have three actions: build, mark for replacement, destruct. To reduce to 2 keys (left and right click), we may add a "destruct" option in the circle menu when left clicking while aiming a buildable marked as replaceable.

**Comment by illwieckz** [comment]
Is it possible to spawn a kind of floating button above the buildable the player can left-click with ckit when the buildable is marked as disposable? it would be the same behavior currently involved when a player aim buildable and mark for deconstruct.

**Comment by slipher** [comment]
Also we should add chat binds to the tutorial. Being unable to talk to the other players is one of the most frustrating things, I think.

**Comment by illwieckz** [comment]
Good point, a simple sentence like "press T to talk with all players, Y with your team, U to an admin" would do the job.

Also, I think since a long time about displaying something like the 5 last messages under the scoreboard so people can press tab to read messages they miss.

**Comment by Zilor** [comment]
>     * [ ]  open server list at startup

I don't think it's contemporary pointing users to the server browser.
Nowadays matchmaking is the preferred way to join servers.
Ofc thinking about a MM system for such a small playerbase is beyond insanity.

But something like Team Fortress 2 old [Quickplay ](https://wiki.teamfortress.com/wiki/Quickplay )system might be fashionable. It provides a simple way to join a server which resambles a MM system.
Searching for the server would be handeld by the client. It simply fetches the serverlist from the master server and then picks the best server based on playercount, ping, etc.

![wot2osDWYELMJPDvD6vvRdk00qJqPqU_ibKcHbLprgU](https://user-images.githubusercontent.com/1292318/83408266-e0b37000-a401-11ea-8ee2-f5569ce88469.png)

**Comment by illwieckz** [comment]
I don't mind if it's contemporary or not, but I mind if it's good for player or not. A “Quick play” or “I’m Feeling Lucky” mechanism seems to be good for player, yes.

For such “I'm Feeling Lucky” mechanism we need to make sure the server selects good servers, so we have to know if a server is good enough, this means we may need to add more public cvars. For example it may be good if, when queried, a server not only answers how many players and bots there is on the server, but also gives some information to guess if the server is properly configured.

Basically, a “good enough server” would be a server with:

- low enough ping
- server not empty (real players takes precedence over bots)
- no password
- there is a working rotation list
- server is likely to be joined quickly (no udp-only download)

for the last point, this would be true if the server:

- only hosts stock paks or
- also hosts custom paks but has http download configured

for the later we may need more information from servers at query time, and maybe more code to build this data.

**Comment by bmorel** [comment]
from the 3 remaining issues:

* interpolation will probably never be implemented. I'm not even sure what this means in that context.
* the dinosaurs vs tremulous thing, I don't understand what it is
* thingy about lan servers seems the only one worth spending time on, except that AFAIK daemon does not have anything to let other computers on a LAN to discover new servers

**Comment by illwieckz** [comment]
> interpolation will probably never be implemented. I'm not even sure what this means in that context.

I added a comment there:  https://github.com/Unvanquished/Unvanquished/issues/609#issuecomment-1230845466

> the dinosaurs vs tremulous thing, I don't understand what it is

It means we cannot assume someone have played Tremulous when we explain the game. Since that issue was opened many works were done on that topic, like in the builtin game guide.

For example in https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/pull/36 I modified the game guide that had as first sentence “In this section you will find a summary of major gameplay changes since Tremulous GPP.” and instead added information new players not having played Tremulous should know, and moved some remaining useful information for Tremulous veterans in latest tab.

> thingy about lan servers seems the only one worth spending time on, except that AFAIK daemon does not have anything to let other computers on a LAN to discover new servers

Dæmon does broadcast queries on LAN to discover nearby servers. The idea is to not have one Internet tab and one LAN tab but a single list of servers, and when refreshing the list, both query the LAN (we can add a checkbox to disable it) and query the masters server. And if there was a LAN server, list it above the Internet servers. Quetoo does something like that it works well.

**Comment by illwieckz** [comment]
About “reference tremulous in obscure places for those dinosaurs having lived that age, no more”,

Along side this being done:

- https://github.com/UnvanquishedAssets/unvanquished_src.dpkdir/pull/36

There was also this being done:

- https://github.com/Unvanquished/Unvanquished/pull/1879

And a small sentence at the top of the web site describing the game without requiring Tremulous knowledge was added recently, plus the other sentences above the download button on home page was added since this ticket was made: https://unvanquished.net/

So let's consider this task as being completed. Future edits would be considered as improving the implemented feature, not implementing the requested feature.

**Comment by illwieckz** [comment]
LAN servers listed atop Internet server request can wait a little, a dedicated thread exists anyway:

- https://github.com/Unvanquished/Unvanquished/issues/1112

For wallwalk interpolation or any other improvement over wallwalk auto pitch, there is a dedicated thread:

- https://github.com/Unvanquished/Unvanquished/issues/609

Let's consider this issue as fixed with future improvements being possible.
